src/collections/aggregate/integration.test.ts

```diff
@@ -0,0 +1,382 @@
+      returnMetrics: collection.metrics
+        .aggregate('text')
+        .text(['count', 'topOccurrencesOccurs', 'topOccurrencesValue']),
+    });
+    expect(result.totalCount).toEqual(100);
+    expect(result.properties.text.count).toEqual(100);
+    expect((result.properties.text as AggregateText).topOccurrences![0].occurs).toEqual(100);
+    expect((result.properties.text as AggregateText).topOccurrences![0].value).toEqual('test');
+  });
+
+  it('should aggregate data without a search and all property metrics', async () => {
+    const result = await collection.aggregate.overAll({
+      returnMetrics: [
+        collection.metrics.aggregate('text').text(['count', 'topOccurrencesOccurs', 'topOccurrencesValue']),
+        collection.metrics.aggregate('texts').text(['count', 'topOccurrencesOccurs', 'topOccurrencesValue']),
+        collection.metrics
+          .aggregate('int')
+          .integer(['count', 'maximum', 'mean', 'median', 'minimum', 'mode', 'sum']),
+        collection.metrics
+          .aggregate('ints')
+          .integer(['count', 'maximum', 'mean', 'median', 'minimum', 'mode', 'sum']),
+        collection.metrics
+          .aggregate('number')
+          .number(['count', 'maximum', 'mean', 'median', 'minimum', 'mode', 'sum']),
+        collection.metrics
+          .aggregate('numbers')
+          .number(['count', 'maximum', 'mean', 'median', 'minimum', 'mode', 'sum']),
+        collection.metrics.aggregate('date').date(['count', 'maximum', 'median', 'minimum', 'mode']),
+        collection.metrics.aggregate('dates').date(['count', 'maximum', 'median', 'minimum', 'mode']),
+        collection.metrics
+          .aggregate('boolean')
+          .boolean(['count', 'percentageFalse', 'percentageTrue', 'totalFalse', 'totalTrue']),
+        collection.metrics
+          .aggregate('booleans')
+          .boolean(['count', 'percentageFalse', 'percentageTrue', 'totalFalse', 'totalTrue']),
+        // Metrics.aggregate('ref').reference(['pointingTo'])
+      ],
+    });
+    // const resDef = await collection.aggregate.overAll({
+    //   returnMetrics: [
+    //     collection.metrics.aggregate('text').text(),
+    //     collection.metrics.aggregate('texts').text(),
+    //     collection.metrics.aggregate('int').integer(),
+    //     collection.metrics.aggregate('ints').integer(),
+    //     collection.metrics.aggregate('number').number(),
+    //     collection.metrics.aggregate('numbers').number(),
+    //     collection.metrics.aggregate('date').date(),
+    //     collection.metrics.aggregate('dates').date(),
+    //     collection.metrics.aggregate('boolean').boolean(),
+    //     collection.metrics.aggregate('booleans').boolean(),
+    //     // Metrics.aggregate('ref').reference(['pointingTo'])
+    //   ],
+    // });
+    // expect(resDef).toEqual(result); // leads to flakes due to weaviate's behaviour when calculating modes of dates
```

Comment:
Is this still needed?

Reply:
Good spot, will remove! I left to identify the flake but it's very low prio